noir-projects/aztec-nr/aztec/src/encrypted_logs/log_assembly_strategies/default_aes128/note.nr

```diff
@@ -201,12 +203,7 @@ pub(crate) fn get_arr_of_size__log_bytes__from_PT<let PT: u32>() -> [u8; (((PT +
 /********************************************************/
 
 // TODO: it feels like this existence check is in the wrong place. In fact, why is it needed at all? Under what circumstances have we found a non-existent note being emitted accidentally?
-fn assert_note_exists<Note, let N: u32>(context: PrivateContext, note: Note)
-where
-    Note: NoteInterface<N>,
-{
-    let note_header = note.get_header();
-    let note_hash_counter = note_header.note_hash_counter;
+fn assert_note_exists<let N: u32>(context: PrivateContext, note_hash_counter: u32) {
```

Comment:
I feel like we could just go ahead and nuke this check because it's only used in 2 places in this file and in those situations the note is always wrapped in `NoteEmission`. So this check would be valuable only if the user was actively trying to use the Aztec.nr API in a non-standard way and instantiate NoteEmission manually and of a note that has not been emitted.

WDYT?

Reply:
Not yet, we'll visit this once we do #8589.